### PR TITLE
[Backup] `az backup`: Add support for MAB backup management type in some sub commands

### PR DESCRIPTION
**Pull request by akshayneema**
**Description**

- Added support for MAB backup management type in below CLI commands -
  - az backup container list
  - az backup container show
  - az backup item list
  - az backup item show
  - az backup job list

- Improved sample examples

- Added support for MAB backup management type in az backup container unregister [code commented for now]

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Backup

File: src/azure-cli/azure/cli/command_modules/backup/_client_factory.py [hunk 1]
```diff
@@ -50,6 +50,10 @@ def vaults_cf(cli_ctx, *_):
     return _common_client_factory(cli_ctx).vaults
 
 
+# def registered_identities_cf(cli_ctx, *_):
+#    return _common_client_factory(cli_ctx).registered_identities
+
+
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/backup/_client_factory.py:56]: Why do wee need to add this block?

Comment (akshayneema) [reply src/azure-cli/azure/cli/command_modules/backup/_client_factory.py:56]: In PR description third point I have added this. I've commented some code as we were not able to test it properly. We will plan to take this with the next release so I've just commented that code.

File: src/azure-cli/azure/cli/command_modules/backup/custom_base.py [hunk 2]
```diff
@@ -298,17 +298,30 @@ def initialize_protectable_items(client, resource_group_name, vault_name, contai
...
+    #    if container_friendly_name is None:
+    #        containrs_client = backup_protection_containers_cf(cmd.cli_ctx)
+    #        container = show_container(cmd, containrs_client, container_name, resource_group_name, vault_name,
+    #                                   backup_management_type)
+    #        container_friendly_name = container.properties.friendly_name
+    #    mab_client = registered_identities_cf(cmd.cli_ctx)
+    #    result = mab_client.delete(resource_group_name, vault_name, container_friendly_name,
+    #                               cls=custom_help.get_pipeline_response)
+    #    return custom_help.track_mab_unregister_operation(cmd.cli_ctx, result, vault_name, resource_group_name,
+    #                                                container_name)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/backup/custom_base.py:324]: May I ask why there are many comment codes in this PR?

**Comment by zhoxing-ms** [comment]
Could you add some tests for those changes?

**Comment by akshayneema** [comment]
> Could you add some tests for those changes?

We have only added support for some GET commands for a backup management type. I will need to use an existing setup to write the test as we don't support creating it from CLI. Is that fine?

File: src/azure-cli/azure/cli/command_modules/backup/custom_base.py (hunk 2 above)
Comment (akshayneema) [reply src/azure-cli/azure/cli/command_modules/backup/custom_base.py:324]: In PR description third point I have added this. I've commented some code as we were not able to test it properly. We will plan to take that with the next release so I've just commented that code.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**